### ISSUE TITLE
bug: npm.jsr.io compiles incorrectly with mixed .ts and .js(+.d.ts) repositories

### ISSUE THREAD

**Issue by mizchi** [jsr-io/jsr#491]
## tl;dr

- When resolving modules with npm.jsr.io as backend, when `.ts` and `.js` are mixed, only `.ts` is expanded to `_dist/*`, but this is out of relative path with `.js`.

Can be reproduced with the package `@mizchi/repro-jsr-npm`.

```
$ pnpm init
$ pnpm add @jsr/mizchi__repro-jsr-npm@0.0.2
```

## Reproduce

```
$ tree .
.
README.md
├── deno.json
├── mod.ts
└── sub
    ├─ lib.d.ts
    └─ lib.js
```

Published by this command.

```
deno publish --allow-slow-types
```

Error on `_dist` relative path

```bash
$ mkdir xxx
$ cd xxx
$ echo "registry=https://npm.jsr.io" > .npmrc
$ pnpm init
$ pnpm add @jsr/mizchi__repro-jsr-npm
$ tree node_modules/@jsr/mizchi__repro-jsr-npm
node_modules/@jsr/mizchi__repro-jsr-npm
├── _dist
│   ├── mod.js
│   └── mod.js.map
├── deno.json
├── mod.ts
├── package.json
└── sub
    ├── lib.d.ts
    └── lib.js
$ cat node_modules/@jsr/mizchi__repro-jsr-npm/_dist/mod.js
export * from "./sub/lib.js"; // import x from "./sub/lib.js";
//# sourceMappingURL=mod.js.map%

```

`_dist/mod.js` can not resolve uncompiled `. /sub/lib.js`

## Context

When building rust code with wasm-pack, it generates `pkg/*.js` and `pkg/*.d.ts`. wasm-pack cannot compile directly to TypeScript.

I ran into this bug when I tried to re-export it from the top-level mod.ts and check if it is importable by node. This seems to affect all relative path resolution when mixing .js, not just wasm-pack products.

(Not related to this case, but in the end node.js could not resolve the wasm binary because it did not support local fetch. This is another problem with node.js)
